### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by Ru-T**
## What Changed?
- Created users table and routes, including root
- Added devise gem and authentication layer
- Added login/logout links to header
- Wrote integration tests for sign in, sign out, sign up
## Why?
- To require users to sign up and log in to use Ruti-Twitter

### CODE REVIEW

**Comment by dkniffin** [comment]
Looks like `.DS_store`, and `app/.DS_Store` got committed. You probably want to add those to the project .gitignore, as well as your personal .gitignore.

File: Gemfile [hunk 1]
```diff
@@ -17,7 +18,7 @@ gem 'coffee-rails', '~> 4.1.0'
 # Use jquery as the JavaScript library
 gem 'jquery-rails'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
-gem 'jbuilder', '~> 2.0'
+#gem 'jbuilder', '~> 2.0'
```
Comment (dkniffin) [inline Gemfile]: You could just remove this line. I think the consensus around the office is that commented code a bad practice.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show]
+
+  def show
+  end
```
Comment (dkniffin) [inline app/controllers/users_controller.rb]: Empty controller methods can be deleted. Rails will render the view regardless.

File: config/cucumber.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<%
+rerun = File.file?('rerun.txt') ? IO.read('rerun.txt') : ""
+rerun_opts = rerun.to_s.strip.empty? ? "--format #{ENV['CUCUMBER_FORMAT'] || 'progress'} features" : "--format #{ENV['CUCUMBER_FORMAT'] || 'pretty'} #{rerun}"
+std_opts = "--format #{ENV['CUCUMBER_FORMAT'] || 'pretty'} --strict --tags ~@wip"
+%>
+default: <%= std_opts %> features
+wip: --tags @wip:3 --wip features
+rerun: <%= rerun_opts %> --format rerun --out rerun.txt --strict --tags ~@wip
```
Comment (dkniffin) [inline config/cucumber.yml]: Hmm. I've never seen this file before. What's it for?

File: db/seeds.rb [hunk 4]
```diff
@@ -5,3 +5,6 @@
 #
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
 #   Mayor.create(name: 'Emanuel', city: cities.first)
```
Comment (dkniffin) [inline db/seeds.rb]: Feel free to delete auto-generated comments like this. They tend to clutter up the code.

File: Gemfile (hunk 1 above)
Comment (Ru-T) [inline Gemfile]: Removed.

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (Ru-T) [inline app/controllers/users_controller.rb]: Removed.

File: db/seeds.rb (hunk 4 above)
Comment (Ru-T) [inline db/seeds.rb]: Deleted!

**Comment by jahammo2** [comment]
I gave @Ru-T the :shipit: in Slack

**Pull request merged**